src/database_migration.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseMigration class to manage database migrations."""
+
+import logging
+from typing import Literal, cast
+
+import ops
+from ops.pebble import ExecError
+
+from constants import FLASK_STATE_DIR
+from exceptions import CharmConfigInvalidError
+
+logger = logging.getLogger(__name__)
+
+
+DatabaseMigrationStatus = Literal["PENDING", "COMPLETED", "FAILED"]
```

Comment:
Also discussed with @gregory-schiano, because our tooling already enforces descriptions on class attributes and because enums are a general concept, we will propose a standard for the team to use enums instead of literals in these cases so we have a consistent approach across the team